pkg/cortex/cortex.go

```diff
@@ -291,6 +294,12 @@ func (t *Cortex) setupThanosTracing() {
 
 // Run starts Cortex running, and blocks until a Cortex stops.
 func (t *Cortex) Run() error {
+	if !t.ModuleManager.IsUserVisibleModule(t.Cfg.Target) {
+		level.Warn(util.Logger).Log(
+			"msg", fmt.Sprintf("'%v' is not a user visible module, is this intended?", t.Cfg.Target),
+			"user-visible-modules", strings.Join(t.ModuleManager.UserVisibleModuleNames(), ", "))
```

Comment:
I would rephrase the message slightly, and drop listing of modules -- it is long and obscures main message.

```suggestion
		level.Warn(util.Logger).Log("msg", "selected target is an internal module, is this intended?", "target", t.Cfg.Target)
```

Reply:
Bike-shedding can be fun and educational when it does not involve money ;) 

Regarding the warning message, I can agree to that since whoever see the warning message either 1) knows what they are doing or 2) will check documentation to find the correct target name.